pkg/vm/emit/emit.go

```diff
@@ -32,20 +33,35 @@ func Bool(w *io.BinWriter, ok bool) {
 	Opcode(w, opcode.PUSHF)
 }
 
+func padRight(s int, buf []byte) []byte {
+	if len(buf) == s {
+		return buf
+	}
+	newBuf := make([]byte, s)
+	copy(newBuf, buf)
+	return newBuf
+}
+
 // Int emits a int type to the given buffer.
 func Int(w *io.BinWriter, i int64) {
 	switch {
 	case i == -1:
 		Opcode(w, opcode.PUSHM1)
-	case i == 0:
-		Opcode(w, opcode.PUSHF)
-	case i > 0 && i < 16:
+	case i >= 0 && i < 16:
 		val := opcode.Opcode(int(opcode.PUSH1) - 1 + int(i))
 		Opcode(w, val)
 	default:
-		bInt := big.NewInt(i)
-		val := IntToBytes(bInt)
-		Bytes(w, val)
+		buf := IntToBytes(big.NewInt(i))
+		l := len(buf)
+		if l > 32 {
```

Comment:
I don't think it could happen with `int64`.

Reply:
Fixed.